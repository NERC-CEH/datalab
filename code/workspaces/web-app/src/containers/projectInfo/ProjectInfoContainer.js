import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import PromisedContentWrapper from '../../components/common/PromisedContentWrapper';
import useCurrentProject from '../../hooks/useCurrentProject';

  shouldComponentUpdate(nextProps) {
    const isFetching = nextProps.currentProject.fetching;
    return !isFetching || this.props.currentProject.isFetching !== isFetching;
const useStyles = makeStyles(theme => ({
  projectTitle: {
    marginRight: theme.spacing(2),
  },
  projectTitleWrapper: {
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'baseline',
  },
  projectKey: {
    marginBottom: theme.typography.h5.marginBottom,
  },
  collaborationLink: {
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'baseline',
    marginTop: theme.spacing(2),
  },
  collaborationLinkHeading: {
    marginRight: theme.spacing(2),
    textTransform: 'uppercase',
    fontSize: '0.9rem',
    fontWeight: 'bold',
    lineHeight: theme.typography.body1.lineHeight,
    letterSpacing: '0.08em',
  },
}));

const ProjectInfoContainer = () => {
  const currentProject = useCurrentProject();
  return <PureProjectInfoContainer currentProject={currentProject} />;
};

export const PureProjectInfoContainer = ({ currentProject }) => {
  const classes = useStyles();
  const { name, key, description, collaborationLink } = currentProject.value;

  return (
    <PromisedContentWrapper promise={currentProject}>
      <div>
        <div className={classes.projectTitleWrapper}>
          <Typography variant="h5" className={classes.projectTitle}>{name}</Typography>
          <Typography variant="projectKey" className={classes.projectKey}>({key})</Typography>
        </div>
        <Typography variant="body1">{description}</Typography>
        <CollaborationLink link={collaborationLink}/>
      </div>
    </PromisedContentWrapper>
  );
};

export const CollaborationLink = ({ link }) => {
  const classes = useStyles();

  if (!link) {
    return null;
  }

  return (
    <div className={classes.collaborationLink}>
      <Typography className={classes.collaborationLinkHeading}>Collaboration Link</Typography>
      <Typography variant="body1">
        <a href={link} target="_blank" rel="noopener noreferrer" >{link}</a>
      </Typography>
    </div>
  );
};

export default ProjectInfoContainer;
