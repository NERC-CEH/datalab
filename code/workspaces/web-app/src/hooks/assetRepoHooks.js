import useShallowSelector from './useShallowSelector';
import assetRepoSelectors from '../selectors/assetRepoSelectors';
import sortByName from '../components/common/sortByName';
import { PUBLIC } from '../components/assetRepo/assetVisibilities';

const useAssetRepoUnsorted = () => useShallowSelector(assetRepoSelectors.assetRepo);

export const useAssetRepo = () => {
  const assetRepo = useAssetRepoUnsorted();
  return {
    ...assetRepo,
    value:
    {
      ...assetRepo.value,
      assets: sortByName(assetRepo.value.assets),
    },
  };
};

const assetVisible = (projectKey, asset) => asset.fileLocation
  && (asset.visible === PUBLIC
  || (asset.projects && asset.projects.map(project => project.key).includes(projectKey)));

export const useVisibleAssets = (projectKey) => {
  const assetRepo = useAssetRepoUnsorted();
  const visibleAssets = assetRepo.value.assets
    ? assetRepo.value.assets.filter(asset => assetVisible(projectKey, asset))
    : [];
  return {
    ...assetRepo,
    value:
    {
      ...assetRepo.value,
      assets: sortByName(visibleAssets),
    },
  };
};
