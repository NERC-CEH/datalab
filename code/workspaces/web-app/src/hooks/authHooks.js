import useShallowSelector from './useShallowSelector';
import authSelectors from '../selectors/authSelectors';

const { currentUserId, currentUserPermissions, currentUserTokens } = authSelectors;

export const useCurrentUserId = () => useShallowSelector(currentUserId);
export const useCurrentUserPermissions = () => useShallowSelector(currentUserPermissions);
export const useCurrentUserTokens = () => useShallowSelector(currentUserTokens);
