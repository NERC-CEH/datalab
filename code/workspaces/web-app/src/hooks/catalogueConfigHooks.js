import useShallowSelector from './useShallowSelector';
import catalogueConfigSelectors from '../selectors/catalogueConfigSelectors';

export const useCatalogueConfig = () => useShallowSelector(catalogueConfigSelectors.catalogueConfig);
export const useCatalogueAvailable = () => useShallowSelector(catalogueConfigSelectors.catalogueAvailable);
