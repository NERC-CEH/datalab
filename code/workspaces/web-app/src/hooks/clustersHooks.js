import useShallowSelector from './useShallowSelector';
import clustersSelectors from '../selectors/clustersSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useClusters = () => useShallowSelector(clustersSelectors.clusters);
export const useClustersByType = (type) => {
  const clusters = useClusters();

  if (!type) {
    return clusters;
  }

  return {
    ...clusters,
    value: clusters.value.filter(cluster => cluster.type === type),
  };
};
