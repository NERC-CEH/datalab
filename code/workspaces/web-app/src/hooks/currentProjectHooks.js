import useShallowSelector from './useShallowSelector';
import currentProjectSelectors from '../selectors/currentProjectSelectors';

const { currentProject, currentProjectKey } = currentProjectSelectors;

export const useCurrentProject = () => useShallowSelector(currentProject);
export const useCurrentProjectKey = () => useShallowSelector(currentProjectKey);
