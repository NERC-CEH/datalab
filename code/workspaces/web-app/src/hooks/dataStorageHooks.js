import useShallowSelector from './useShallowSelector';
import dataStorageSelectors from '../selectors/dataStorageSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useDataStorageArray = () => useShallowSelector(dataStorageSelectors.dataStorageArray);
export const useDataStorageForUserInProject = (userId, projectKey) => {
  const dataStores = useDataStorageArray();
  return {
    ...dataStores,
    value: dataStores.value.filter(store => store.projectKey === projectKey && store.users.includes(userId)),
  };
};
