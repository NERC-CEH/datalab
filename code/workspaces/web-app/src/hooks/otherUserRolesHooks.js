import useShallowSelector from './useShallowSelector';
import otherUserRolesSelectors from '../selectors/otherUserRolesSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useOtherUserRoles = () => useShallowSelector(otherUserRolesSelectors.otherUserRoles);
