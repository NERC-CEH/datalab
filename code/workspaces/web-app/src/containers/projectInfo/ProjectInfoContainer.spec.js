import React from 'react';
import { shallow } from 'enzyme';
import ProjectInfoContainer, { PureProjectInfoContainer, CollaborationLink } from './ProjectInfoContainer';
import { useCurrentProject } from '../../hooks/currentProjectHooks';

jest.mock('../../hooks/currentProjectHooks');

const testProj = {
  key: 'testproj',
  name: 'Test Project',
  description: 'A description about Test Project.',
  collaborationLink: 'https://testlab.test-datalabs.nerc.ac.uk',
};

const currentProject = { fetching: false, value: testProj };

useCurrentProject.mockReturnValue(currentProject);

describe('ProjectInfoContainer', () => {
  it('renders correctly passing correct props to children', () => {
    expect(shallow(<ProjectInfoContainer/>)).toMatchSnapshot();
  });
});

describe('PureProjectInfoContainer', () => {
  describe('renders correctly to match snapshot', () => {
    it('when there is a collaboration link', () => {
      expect(
        shallow(<PureProjectInfoContainer currentProject={currentProject} />),
      ).toMatchSnapshot();
    });

    it('when there is not a collaboration link', () => {
      const project = {
        ...currentProject,
        value: {
          ...testProj,
          collaborationLink: undefined,
        },
      };
      expect(
        shallow(<PureProjectInfoContainer currentProject={project} />),
      ).toMatchSnapshot();
    });
  });
});

describe('CollaborationLink', () => {
  describe('renders correctly to match snapshot', () => {
    it('when there is a collaboration link', () => {
      expect(
        shallow(<CollaborationLink link={testProj.collaborationLink} />),
      ).toMatchSnapshot();
    });

    it('when there is not a collaboration link', () => {
      expect(
        shallow(<CollaborationLink link={undefined} />),
      ).toMatchSnapshot();
    });
  });
});
