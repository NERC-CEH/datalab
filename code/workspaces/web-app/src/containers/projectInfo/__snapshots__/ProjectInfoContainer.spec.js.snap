// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollaborationLink renders correctly to match snapshot when there is a collaboration link 1`] = `
<div
  className="makeStyles-collaborationLink-5"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-collaborationLinkHeading-6"
  >
    Collaboration Link
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    variant="body1"
  >
    <a
      href="https://testlab.test-datalabs.nerc.ac.uk"
      rel="noopener noreferrer"
      target="_blank"
    >
      https://testlab.test-datalabs.nerc.ac.uk
    </a>
  </WithStyles(ForwardRef(Typography))>
</div>
`;

exports[`CollaborationLink renders correctly to match snapshot when there is not a collaboration link 1`] = `""`;

exports[`ProjectInfoContainer renders correctly passing correct props to children 1`] = `
<PureProjectInfoContainer
  currentProject={
    Object {
      "fetching": false,
      "value": Object {
        "collaborationLink": "https://testlab.test-datalabs.nerc.ac.uk",
        "description": "A description about Test Project.",
        "key": "testproj",
        "name": "Test Project",
      },
    }
  }
/>
`;

exports[`PureProjectInfoContainer renders correctly to match snapshot when there is a collaboration link 1`] = `
<PromisedContentWrapper
  promise={
    Object {
      "fetching": false,
      "value": Object {
        "collaborationLink": "https://testlab.test-datalabs.nerc.ac.uk",
        "description": "A description about Test Project.",
        "key": "testproj",
        "name": "Test Project",
      },
    }
  }
>
  <div>
    <div
      className="makeStyles-projectTitleWrapper-3"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-projectTitle-1"
        variant="h5"
      >
        Test Project
      </WithStyles(ForwardRef(Typography))>
      <ResourceInfoSpan
        className="makeStyles-ResourceInfoSpan-4"
      >
        (testproj)
      </ResourceInfoSpan>
    </div>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-description-2"
      variant="body1"
    >
      A description about Test Project.
    </WithStyles(ForwardRef(Typography))>
    <CollaborationLink
      link="https://testlab.test-datalabs.nerc.ac.uk"
    />
    <div
      className="makeStyles-userBlock-7"
    >
      <div>
        <UserBlock
          classes={
            Object {
              "ResourceInfoSpan": "makeStyles-ResourceInfoSpan-4",
              "chipTitle": "makeStyles-chipTitle-8",
              "collaborationLink": "makeStyles-collaborationLink-5",
              "collaborationLinkHeading": "makeStyles-collaborationLinkHeading-6",
              "description": "makeStyles-description-2",
              "projectTitle": "makeStyles-projectTitle-1",
              "projectTitleWrapper": "makeStyles-projectTitleWrapper-3",
              "userBlock": "makeStyles-userBlock-7",
            }
          }
          name="admin"
          users={Array []}
        />
        <UserBlock
          classes={
            Object {
              "ResourceInfoSpan": "makeStyles-ResourceInfoSpan-4",
              "chipTitle": "makeStyles-chipTitle-8",
              "collaborationLink": "makeStyles-collaborationLink-5",
              "collaborationLinkHeading": "makeStyles-collaborationLinkHeading-6",
              "description": "makeStyles-description-2",
              "projectTitle": "makeStyles-projectTitle-1",
              "projectTitleWrapper": "makeStyles-projectTitleWrapper-3",
              "userBlock": "makeStyles-userBlock-7",
            }
          }
          name="user"
          users={
            Array [
              Object {
                "name": "user1@test",
                "role": "user",
                "userId": "user1",
              },
            ]
          }
        />
        <UserBlock
          classes={
            Object {
              "ResourceInfoSpan": "makeStyles-ResourceInfoSpan-4",
              "chipTitle": "makeStyles-chipTitle-8",
              "collaborationLink": "makeStyles-collaborationLink-5",
              "collaborationLinkHeading": "makeStyles-collaborationLinkHeading-6",
              "description": "makeStyles-description-2",
              "projectTitle": "makeStyles-projectTitle-1",
              "projectTitleWrapper": "makeStyles-projectTitleWrapper-3",
              "userBlock": "makeStyles-userBlock-7",
            }
          }
          name="viewer"
          users={Array []}
        />
      </div>
    </div>
  </div>
</PromisedContentWrapper>
`;

exports[`PureProjectInfoContainer renders correctly to match snapshot when there is not a collaboration link 1`] = `
<PromisedContentWrapper
  promise={
    Object {
      "fetching": false,
      "value": Object {
        "collaborationLink": undefined,
        "description": "A description about Test Project.",
        "key": "testproj",
        "name": "Test Project",
      },
    }
  }
>
  <div>
    <div
      className="makeStyles-projectTitleWrapper-3"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-projectTitle-1"
        variant="h5"
      >
        Test Project
      </WithStyles(ForwardRef(Typography))>
      <ResourceInfoSpan
        className="makeStyles-ResourceInfoSpan-4"
      >
        (testproj)
      </ResourceInfoSpan>
    </div>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-description-2"
      variant="body1"
    >
      A description about Test Project.
    </WithStyles(ForwardRef(Typography))>
    <CollaborationLink />
    <div
      className="makeStyles-userBlock-7"
    >
      <div>
        <UserBlock
          classes={
            Object {
              "ResourceInfoSpan": "makeStyles-ResourceInfoSpan-4",
              "chipTitle": "makeStyles-chipTitle-8",
              "collaborationLink": "makeStyles-collaborationLink-5",
              "collaborationLinkHeading": "makeStyles-collaborationLinkHeading-6",
              "description": "makeStyles-description-2",
              "projectTitle": "makeStyles-projectTitle-1",
              "projectTitleWrapper": "makeStyles-projectTitleWrapper-3",
              "userBlock": "makeStyles-userBlock-7",
            }
          }
          name="admin"
          users={Array []}
        />
        <UserBlock
          classes={
            Object {
              "ResourceInfoSpan": "makeStyles-ResourceInfoSpan-4",
              "chipTitle": "makeStyles-chipTitle-8",
              "collaborationLink": "makeStyles-collaborationLink-5",
              "collaborationLinkHeading": "makeStyles-collaborationLinkHeading-6",
              "description": "makeStyles-description-2",
              "projectTitle": "makeStyles-projectTitle-1",
              "projectTitleWrapper": "makeStyles-projectTitleWrapper-3",
              "userBlock": "makeStyles-userBlock-7",
            }
          }
          name="user"
          users={
            Array [
              Object {
                "name": "user1@test",
                "role": "user",
                "userId": "user1",
              },
            ]
          }
        />
        <UserBlock
          classes={
            Object {
              "ResourceInfoSpan": "makeStyles-ResourceInfoSpan-4",
              "chipTitle": "makeStyles-chipTitle-8",
              "collaborationLink": "makeStyles-collaborationLink-5",
              "collaborationLinkHeading": "makeStyles-collaborationLinkHeading-6",
              "description": "makeStyles-description-2",
              "projectTitle": "makeStyles-projectTitle-1",
              "projectTitleWrapper": "makeStyles-projectTitleWrapper-3",
              "userBlock": "makeStyles-userBlock-7",
            }
          }
          name="viewer"
          users={Array []}
        />
      </div>
    </div>
  </div>
</PromisedContentWrapper>
`;
